src/styles/tools/_collections.scss

```diff
@@ -72,6 +97,38 @@
     }
 }
 
+// Mixin to generate CSS custom properties for links theme.
+//
+// 1. Links theming is only available for feedback colors.
+//
+// @param {String} $prefix - The prefix for the CSS custom properties.
+// @param {String} $variant-value - The value of the variant.
+//
+// Example:
+//
+// @include generate-link-properties(
+//     $prefix: "rui-",
+//     $variant-value: "success",
+// );
+//
+// … will output:
+//
+// --rui-local-link-color: var(--rui-color-feedback-success);
+// --rui-local-link-color-hover: var(--rui-color-feedback-success-hover);
+// --rui-local-link-color-active: var(--rui-color-feedback-success-active);
+
+@mixin generate-link-properties($prefix, $variant-value) {
+    $color-category: _get-category-by-value($value: $variant-value, $collections: collections.$colors);
+
+    // 1.
+    // TODO: Figure out what to do with the `dark` color category where blue links do not work.
```

Comment:
This seems to be an issue. As stated in #492, we want to keep the basic appearance for links inside components in the Neutral Light color. However, Neutral Dark is unreadable then.

I have to think about it yet. 🤔 

<img width="785" alt="obrazek" src="https://github.com/user-attachments/assets/d50f21ef-88e4-4e92-b99d-37ce01b1e620">

Reply:
Personal opinion: Inherit color for neutral dark color (and keep only neutral light color) OR split link color property/ies between light and dark variant/s. This is beyond this discussion, but if we generally split colors between light and dark, couldn't we use it later for dark mode variant (which we haven't talk about it yet)?

Reply:
Light and dark colors (the neutral colors collection) and light and dark modes are something different. But if you mean we would face the same problem during dark mode design, you are probably right. Just with the difference that light and dark neutral colors will be both somewhat dark, just using a different shade of gray.

When I'm thinking about it from the design perspective, it makes sense to use the default link appearance in light color variants. Even if it means an exception in our code.

(Other way would be adding even more design tokens (= custom properties) which sounds like an overkill…)

Reply:
Maybe we can have another exception for `Alert`s where dark text would work well 😄 

<img width="851" alt="obrazek" src="https://github.com/user-attachments/assets/51655fae-07bf-4c71-9b1c-ff701ee1ef12">

Reply:
I decided to keep the exception only for links inside `Card`.

> ℹ️ To meet users' expectation, links in the light neutral variant of Card use the [default link color](http://localhost:8000/docs/foundation/colors/#text-colors). In other variants, links are colored according to the variant.

![image](https://github.com/user-attachments/assets/40241d3f-ec8a-42b8-b3ab-822699352980)
![image](https://github.com/user-attachments/assets/410f3536-2128-45d2-9af1-a93059e81710)

Reply:
I think that for  `Alert` and `Card` we should either use default link color in both in `light` and `dark` versions or we should use default link color for `light` and text color for `dark` versions.

Unifying the behavior across components would make it much easier to grasp and it is a better DX. Tweaking the colors and boldness to make it work can then be performed by each RUI based project.

Maybe, if possible, we can tweak the link color to make it work in both color contexts.

---


Presently the solution with different visualisation of `Alert` and `Card` works because `Alert` shows link with emphasis. I think this is something we want to be able to turn of, that is `Alert` should accept along with `--rui-Alert__emphasis__font-weight` also something like `--rui-Alert__link__font-weight`. Maybe someone does not like link-as-emphasis approach.

This would allow us to render `Alert` with normal weight links. Then there would be no reason to have differently link color between `Alert` and `Card`.

Reply:
> Maybe, if possible, we can tweak the link color to make it work in both color contexts.

I tried and I don't see a way here…

> Then there would be no reason to have differently link color between Alert and Card.

I changed it and there is currently no exception. It looks like this which we know is not ideal, so I filed a followup #577 for it. But the approach is the same everywhere now which is what we want.

<img width="754" alt="image" src="https://github.com/user-attachments/assets/ce4db291-3b95-44d3-8fdd-8bae3e913168">
<img width="791" alt="image" src="https://github.com/user-attachments/assets/e816e869-6eac-4bdd-bed6-d71452020b57">
<img width="779" alt="image" src="https://github.com/user-attachments/assets/08962415-8640-4b1a-bd63-b2e29210884d">